pkg/agent/testdata/AKSUbuntu1604+Containerd/line70.sh

```diff
@@ -327,6 +327,15 @@ EOF
     echo "${CONTAINERD_CONFIG_CONTENT}" | base64 -d > /etc/containerd/config.toml || exit $ERR_FILE_WATCH_TIMEOUT
   fi
 
+  CONTAINERD_CONFIG_REGISTRY_HOST_MCR="/etc/containerd/certs.d/mcr.microsoft.com/hosts.toml"
+  mkdir -p "$(dirname "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}")"
+  touch "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}"
+  chmod 0644 "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}"
+  tee "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}" > /dev/null <<EOF
```

Comment:
can you see if you can preserve unit test coverage by adding new tests to https://github.com/Azure/AgentBaker/blob/dev/spec/parts/linux/cloud-init/artifacts/cse_config_spec.sh?

we recently started using this so existing examples are a bit lacking unfortunately

Reply:
I tried to add UT, but failed with the permission issue.

I think, we need a container environment to run these UTs and add the required permissions (many functions is cse_config.sh needs to create files).

If there are other ways to fix the permission issue, please let me know and I'm glad to add UTs.

```
Cloning into './shellspecsrc'...
make[1]: Leaving directory '/home/runner/work/AgentBaker/AgentBaker/hack/tools'
Running: /usr/bin/bash [bash 5.1.16(1)-release]
........F.............................

Examples:
  1) cse_config.sh ensureContainerd should generate containerd config correctly
     When call ensureContainerd

     1.1) Example aborted (exit status: 6)

            mkdir: cannot create directory ‘/etc/systemd/system/containerd.service.d’: Permission denied
            tee: /etc/systemd/system/containerd.service.d/exec_start.conf: No such file or directory
            ./parts/linux/cloud-init/artifacts/cse_config.sh: line 3[43](https://github.com/Azure/AgentBaker/actions/runs/12269430636/job/34232973503?pr=5393#step:4:44): /etc/containerd/config.toml: Permission denied

          # spec/parts/linux/cloud-init/artifacts/cse_config_spec.sh:105-108

Finished in 3.148 seconds (user: 3.288s, sys: 0.5[47](https://github.com/Azure/AgentBaker/actions/runs/12269430636/job/34232973503?pr=5393#step:4:48)s) [time: bash-builtin]
38 examples, 1 failure


Failure examples / Errors: (Listed here affect your suite's status)

shellspec spec/parts/linux/cloud-init/artifacts/cse_config_spec.sh:105 # 1) cse_config.sh ensureContainerd should generate containerd config correctly FAILED

make: *** [Makefile:82: shellspec] Error 101
```

Reply:
I also already created an issue to address containerization, because I definitely agree that we should be running these within a container image where we can control the available shell: https://github.com/Azure/AgentBaker/issues/5322

Reply:
added the UTs by mocking